packages/salesforcedx-vscode-apex/src/oas/documentProcessorPipeline/oasValidationStep.ts

```diff
@@ -6,23 +6,62 @@
  */
 
 import { Spectral } from '@stoplight/spectral-core';
+import * as vscode from 'vscode';
 import { ProcessorInputOutput, ProcessorStep } from './processorStep';
 import ruleset from './ruleset.spectral';
 
 export class OasValidationStep implements ProcessorStep {
+  private diagnosticCollection: vscode.DiagnosticCollection;
+  private className: string;
+
+  constructor(className: string) {
```

Comment:
@CristiCanizales The constructor does not need to be changed. The process function. accepts the output from the previous step. On that object is the context element that contains the class name,